css/tabbar.css

```diff
@@ -111,27 +111,27 @@
   height: var(--tab-radius) !important;
   background: var(--back-color1);
   background-clip: content-box;
-  bottom: calc(0px - var(--tab-border-width)) !important;
+  bottom: calc(0rem - var(--tab-border-width)) !important;
   pointer-events: none !important;
   border-bottom: var(--tab-border-width) solid transparent !important;
 }
 
 .tabbrowser-tab[visuallyselected] .tab-stack::before {
   border-bottom-right-radius: var(--tab-radius) !important;
   left: 0 !important;
-  transform: translateX(calc(0px - var(--tab-border-width) - var(--tab-radius))) !important;
+  transform: translateX(calc(0rem - var(--tab-border-width) - var(--tab-radius))) !important;
   border-right: var(--tab-border-width) solid transparent !important;
-  box-shadow: 4px 4px 0 4px var(--back-color1), calc(0px - var(--box-shadow-width) - var(--tab-border-width)) 0 0 var(--box-shadow-width) var(--back-color1) !important;
-  clip-path: polygon(calc(0px - var(--box-shadow-width)) -10px, 100% -10px, 100% 100%, 0 100%, 0 calc(100% - var(--tab-border-width)), calc(0px - var(--box-shadow-width)) calc(100% - var(--tab-border-width)));
+  box-shadow: 0.25rem 0.25rem 0 0.25rem var(--back-color1), calc(0rem - var(--box-shadow-width) - var(--tab-border-width)) 0 0 var(--box-shadow-width) var(--back-color1) !important;
+  clip-path: polygon(calc(0rem - var(--box-shadow-width)) calc(-0.625rem), 100% -0.625rem, 100% 100%, 0 100%, 0 calc(100% - var(--tab-border-width)), calc(0rem - var(--box-shadow-width)) calc(100% - var(--tab-border-width)));
 }
 
 .tabbrowser-tab[visuallyselected] .tab-stack::after {
   border-bottom-left-radius: var(--tab-radius) !important;
   right: 0 !important;
   transform: translateX(calc(var(--tab-border-width) + var(--tab-radius))) !important;
   border-left: var(--tab-border-width) solid transparent !important;
-  box-shadow: -4px 4px 0 4px var(--back-color1), calc(var(--box-shadow-width) + var(--tab-border-width)) 0 0 var(--box-shadow-width) var(--back-color1) !important;
-  clip-path: polygon(var(--box-shadow-width) -10px, 0 -10px, 0 100%, 100% 100%, 100% calc(100% - var(--tab-border-width)), var(--box-shadow-width) calc(100% - var(--tab-border-width)));
+  box-shadow: -0.25rem 0.25rem 0 0.25rem var(--back-color1), calc(var(--box-shadow-width) + var(--tab-border-width)) 0 0 var(--box-shadow-width) var(--back-color1) !important;
+  clip-path: polygon(var(--box-shadow-width) -0.625rem, 0 -0.625rem, 0 100%, 100% 100%, 100% calc(100% - var(--tab-border-width)), var(--box-shadow-width) calc(100% - var(--tab-border-width)));
```

Comment:
> ![](https://user-images.githubusercontent.com/29517049/215238895-2361bf9f-458d-4956-b500-ccf996726006.png)
>
> 2. Some artifacts on the right vertical line of the open tab. Is the next tab's invisible border causing a conflict?
>
> Re: point 2 (right vertical line getting clipped), it seems the rules related to .tabbrowser-tab[visuallyselected] .tab-stack::after are affecting this (especially the clip-path), but even after converting the units to rem, it wasn't fixed. Not sure what to do about it.

Does the changes in `clip-path` incorporate this issue?

Reply:
Nope, as mentioned in the quoted comment, switching to `rem` does _not_ fix the issue. Additionally, it shows up sporadically, depending on the number of open tabs. Maybe some other `px` values are interfering